### ISSUE TITLE
Add stdio halfclose tests

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1742]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1742|
|Type|task|
|Created|2006-05-22 21:45:50Z|

There aren't any now.

<details><summary>Searchable metadata</summary>

```
trac-id__1742 1742
type__task task
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1148334350000000 1148334350000000
changetime__1268010798000000 1268010798000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

(In [16921]) remove file which does not do anything

Author: exarkun
Reviewer: radix
Fixes [#11248](https://github.com/twisted/twisted/issues/11248)
Refs [#1742](https://github.com/twisted/twisted/issues/1742)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [28666]) Add a halfclose stdio test

refs [#4352](https://github.com/twisted/twisted/issues/4352)
refs [#1742](https://github.com/twisted/twisted/issues/1742)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [28668]) Fix the new test in pollreactor and epollreactor

refs [#4352](https://github.com/twisted/twisted/issues/4352)
refs [#1742](https://github.com/twisted/twisted/issues/1742)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [28679]) Merge poll-stdio-halfclose-4352

Author: exarkun
Reviewer: therve
Fixes: [#1742](https://github.com/twisted/twisted/issues/1742)
Fixes: [#4352](https://github.com/twisted/twisted/issues/4352)

Fix support for stdio halfclose in pollreactor, epollreactor, and gtk2reactor.
Also add a test for stdio halfclose that any reactor can run.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
